lntest/miner/miner.go

```diff
@@ -292,15 +293,21 @@ func (h *HarnessMiner) AssertTxInMempool(txid *chainhash.Hash) *wire.MsgTx {
 			return fmt.Errorf("empty mempool")
 		}
 
-		for _, memTx := range mempool {
-			// Check the values are equal.
-			if *memTx == *txid {
-				return nil
-			}
+		isEqual := func(memTx *chainhash.Hash) bool {
+			return *memTx == *txid
 		}
+		result := fn.Find(isEqual, mempool)
 
-		return fmt.Errorf("txid %v not found in mempool: %v", txid,
-			mempool)
+		if result.IsNone() {
+			return fmt.Errorf("txid %v not found in mempool: %v",
+				txid, mempool)
+		}
+
+		result.WhenSome(func(txid *chainhash.Hash) {
+			msgTx = h.GetRawTransaction(txid).MsgTx()
+		})
```

Comment:
Is there any version of `wait.NoError` that allows a real return value (other than just the error)? CC @yyforyongyu

Reply:
recreated one see `NoErrorWithValue` lmk what you think.

Reply:
built it back to the original `NoError` func